jetbrains-core/src/software/aws/toolkits/jetbrains/services/s3/objectActions/CopyPathAction.kt

```diff
@@ -1,30 +1,23 @@
-// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 // SPDX-License-Identifier: Apache-2.0
 
 package software.aws.toolkits.jetbrains.services.s3.objectActions
 
 import com.intellij.icons.AllIcons
-import com.intellij.openapi.actionSystem.AnActionEvent
-import com.intellij.openapi.actionSystem.LangDataKeys
 import com.intellij.openapi.ide.CopyPasteManager
-import com.intellij.openapi.project.DumbAwareAction
-import software.aws.toolkits.jetbrains.services.s3.editor.S3TreeContinuationNode
+import com.intellij.openapi.project.Project
+import software.aws.toolkits.jetbrains.services.s3.editor.S3TreeNode
 import software.aws.toolkits.jetbrains.services.s3.editor.S3TreeTable
 import software.aws.toolkits.jetbrains.services.telemetry.TelemetryConstants.TelemetryResult
 import software.aws.toolkits.jetbrains.services.telemetry.TelemetryService
 import software.aws.toolkits.resources.message
 import java.awt.datatransfer.StringSelection
 
-class CopyPathAction(private val treeTable: S3TreeTable) : DumbAwareAction(message("s3.copy.path"), null, AllIcons.Actions.Copy) {
-    override fun actionPerformed(e: AnActionEvent) {
-        treeTable.getSelectedNodes().firstOrNull()?.let {
-            CopyPasteManager.getInstance().setContents(StringSelection(it.key))
-            TelemetryService.recordSimpleTelemetry(e.getRequiredData(LangDataKeys.PROJECT), TELEMETRY_NAME, TelemetryResult.Succeeded)
-        }
-    }
+class CopyPathAction(private val project: Project, treeTable: S3TreeTable) : SingleS3ObjectAction(treeTable, message("s3.copy.path"), AllIcons.Actions.Copy) {
 
-    override fun update(e: AnActionEvent) {
-        e.presentation.isEnabled = treeTable.selectedRows.size == 1 && !treeTable.getSelectedNodes().any { it is S3TreeContinuationNode }
+    override fun performAction(node: S3TreeNode) {
+        CopyPasteManager.getInstance().setContents(StringSelection(node.key))
+        TelemetryService.recordSimpleTelemetry(project, TELEMETRY_NAME, TelemetryResult.Succeeded)
```

Comment:
@hunterwerlla / @rli thoughts on pushing these `TELEMETRY_NAME` constants into a single place where *all* telemetry constants go (for now until we get a generated version) ?

(obviously this would be a separate PR / change).

Reply:
I like that idea

Reply:
Agree